### PR TITLE
mimxrt/Makefile: Refactors Makefile.

### PR DESCRIPTION
**Pull request by alphaFred**
Refactors grouping of Makefile variables and reorders variable values in
alphabetic order.
Reworks handling of board specific options for SDRAM, SDCARD, flash, and floating point with modifications in the Makefile, linker scripts and board makefiles to simplify over all build scripts.

Signed-off-by: Philipp Ebensberger <philipp.ebensberger@3bricks-software.de>

### CODE REVIEW

File: ports/mimxrt/boards/MIMXRT1011.ld [hunk 1]
```diff
@@ -3,7 +3,7 @@
 reserved_size = MICROPY_HW_FLASH_RESERVED;
 #endif
 
-#if MICROPY_HW_FLASH_TYPE==qspi_nor
+#if MICROPY_HW_FLASH_TYPE==qspi_nor_flash
```
Comment (dpgeorge) [inline ports/mimxrt/boards/MIMXRT1011.ld:6]: put spaces around `==`

File: ports/mimxrt/boards/MIMXRT1052.ld [hunk 2]
```diff
@@ -3,9 +3,9 @@
 reserved_size = MICROPY_HW_FLASH_RESERVED;
 #endif
 
-#if MICROPY_HW_FLASH_TYPE==qspi_nor
+#if MICROPY_HW_FLASH_TYPE==qspi_nor_flash
```
Comment (dpgeorge) [inline ports/mimxrt/boards/MIMXRT1052.ld:6]: spaces around `==`

File: ports/mimxrt/boards/MIMXRT1052.ld [hunk 3]
```diff
@@ -3,9 +3,9 @@
...
 flash_start         = 0x60000000;
-#elif MICROPY_HW_FLASH_TYPE==hyperflash
+#elif MICROPY_HW_FLASH_TYPE==qspi_hyper_flash
```
Comment (dpgeorge) [inline ports/mimxrt/boards/MIMXRT1052.ld:8]: spaces around `==`

**Comment by dpgeorge** [comment]
Thanks, this looks good.

I'm just wondering about the last commit that "fixes formatting".  Can this be squashed into earlier commits, or is it intended to be separate?  Because it changes some `?=` to `=` that's not really fixing formatting, that's changing behaviour.

File: ports/mimxrt/boards/MIMXRT1011.ld (hunk 1 above)
Comment (alphaFred) [reply ports/mimxrt/boards/MIMXRT1011.ld]: fixed

File: ports/mimxrt/boards/MIMXRT1052.ld (hunk 3 above)
Comment (alphaFred) [reply ports/mimxrt/boards/MIMXRT1052.ld]: fixed

**Comment by alphaFred** [comment]
I modified the previous commit and split it into two with descriptive commit message. Additionally I fixed your review findings.

I have two unrelated questions. First of all, I know you have a lot of other more important things at hand but I still have a PR open for adding a bootloader for the mimxrt port (#8229). Is there any chance that this PR will be merged anytime soon or at all? I know that this is a big change which would require a lot of effort from your side. I really would like to see that PR merged so if you could give me any guidance on how I could make the change easier for you to review I would be thrilled. On the other hand if there is no realistic chance for it to be merged please let me know so I will not spend the effort to keep it up to date all the time.

Secondly I used to rebase all the time for the PRs such that there is no merge commit present and then force pushed the chances. I am just wondering if you would be also ok when I would regularly  merge the current main branch  into my PRs to keep them up to date?

**Comment by dpgeorge** [comment]
Merged in fbc50196ad60fa548e79f8d1e5d87c3fa93c7848 through 4ce1adab3153bb0dd5afcd0ddefc219b91d1a13e

**Comment by dpgeorge** [comment]
> I still have a PR open for adding a bootloader for the mimxrt port (#8229). Is there any chance that this PR will be merged anytime soon or at all?

I will review it soon.

> I used to rebase all the time for the PRs such that there is no merge commit present and then force pushed the chances. I am just wondering if you would be also ok when I would regularly merge the current main branch into my PRs to keep them up to date?

No, merging makes it harder to rebase (sometimes).  It's best to just leave it as-is.  A rebase and force push is only needed by you if things change drastically on master, and usually I would ask you to do it.  Otherwise you can just leave a PR as is once you open it.

**Pull request closed without merging**